keras/src/losses/losses.py

```diff
@@ -23,7 +24,9 @@ def __init__(
         self._fn_kwargs = kwargs
 
     def call(self, y_true, y_pred):
-        y_true, y_pred = squeeze_or_expand_to_same_rank(y_true, y_pred)
+        y_true, y_pred = optree.tree_transpose_map(
```

Comment:
Why is this necessary?

Reply:
Do you mean why not simply use `tree_map`?
A simple map would produce a single pytree, but need two for `y_true` and `y_pred`.

Reply:
I removed  `squeeze_or_expand_to_same_rank` altogether, as far as I know there was a `tensorboard` test falling that I fixed.